module/interface.py

```diff
@@ -274,18 +285,119 @@ def codegen(self, folder, project_type='', parameters='vectors',
         cg.codegen(work, folder, python_ext_name, project_type,
                    embedded, force_rewrite, float_flag, long_flag)
 
-
-def solve(P=None, q=None, A=None, l=None, u=None, **settings):
+    def adjoint_derivative(self, dx=None, dy_u=None, dy_l=None,
+            P_idx=None, A_idx=None, diff_mode='lsqr'):
+        """
+        Compute adjoint derivative after solve.
         """
-        Solve problem of the form 
 
-        minimize     1/2 x' * P * x + q' * x
-        subject to   l <= A * x <= u
+        P, q = self._derivative_cache['P'], self._derivative_cache['q']
+        A = self._derivative_cache['A']
+        l, u = self._derivative_cache['l'], self._derivative_cache['u']
+
+        results = self._derivative_cache['results']
```

Comment:
Can we do:
```python
try:
    results = self._derivative_cache['results']
except KeyError:
    raise ValueError("Problem has not been solved. You cannot take derivatives. Please call the solve function.")
```

Reply:
Fixed both. I did not spent too much time fixing these issues since I think they will be much easier to deal with from the C code where the cache will correspond to our workspace.